paddle/fluid/framework/parallel_executor.cc

```diff
@@ -299,6 +299,12 @@ void ParallelExecutor::FeedAndSplitTensorIntoLocalScopes(
 }
 
 ParallelExecutor::~ParallelExecutor() {
+  const auto dev_ctxs =
+      platform::DeviceContextPool::Instance().GetAllDeviceContexts();
```

Comment:
Is this better: `platform::DeviceContextPool::Instance().Get(place_)->Wait(); `?